src/main/java/org/jabref/logic/importer/fetcher/ZbMATH.java

```diff
@@ -40,17 +46,55 @@ public String getName() {
         return "zbMATH";
     }
 
-    /**
-     * TODO: Implement EntryBasedParserFetcher
-     * We use the zbMATH Citation matcher (https://www.zbmath.org/citationmatching/)
-     * instead of the usual search since this tool is optimized for finding a publication based on partial information.
-     */
-    /*
     @Override
     public URL getURLForEntry(BibEntry entry) throws URISyntaxException, MalformedURLException, FetcherException {
-        // Example: https://zbmath.org/citationmatching/match?q=Ratiu
+        Optional<String> zblidInEntry = entry.getField(StandardField.ZBL_NUMBER);
+        if (zblidInEntry.isPresent()) {
+            // zbmath id is already present
+            return getUrlForIdentifier(zblidInEntry.get());
+        }
+
+        URIBuilder uriBuilder = new URIBuilder("https://zbmath.org/citationmatching/match");
+        uriBuilder.addParameter("n", "1"); // return only the best matching entry
+        uriBuilder.addParameter("m", "5"); // return only entries with a score of at least 5
+
+        entry.getFieldOrAlias(StandardField.TITLE).ifPresent(title -> uriBuilder.addParameter("t", title));
+        entry.getFieldOrAlias(StandardField.JOURNAL).ifPresent(journal -> uriBuilder.addParameter("j", journal));
+        entry.getFieldOrAlias(StandardField.YEAR).ifPresent(year -> uriBuilder.addParameter("y", year));
+        entry.getFieldOrAlias(StandardField.PAGINATION)
+             .ifPresent(pagination -> uriBuilder.addParameter("p", pagination));
+        entry.getFieldOrAlias(StandardField.VOLUME).ifPresent(volume -> uriBuilder.addParameter("v", volume));
+        entry.getFieldOrAlias(StandardField.ISSUE).ifPresent(issue -> uriBuilder.addParameter("i", issue));
+
+        if (entry.getFieldOrAlias(StandardField.AUTHOR).isPresent()) {
+            // replace "and" by ";" as citation matching API uses ";" for separation
+            String author = entry.getFieldOrAlias(StandardField.AUTHOR).get();
```

Comment:
Please use the `AuthorList` class for parsing and reformatting the authors. This also handles a few edge cases such as `author={{United and Free States}}` which is one author (an "organization") and not two.

Reply:
Do you mean this class [https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/entry/AuthorList.java](https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/entry/AuthorList.java)?

I need the authors separated by semicolon (lastname before firstname). Is there a function doing this? I have not found one. If it not exists, I guess I have to add one, similar to getAsLastFirstNamesWithAnd(), replacing " and " in line 480 by "; "?